src/main/java/runhistoryplus/patches/PotionUseAndDiscardRunHistoryPatch.java

```diff
@@ -179,4 +251,98 @@ public int[] Locate(CtBehavior ctMethodToPatch) throws CannotCompileException, P
             }
         }
     }
+
+    public static void obtainPotionAlchemizeOrEntropicBrew(final AbstractPotion potion, String source) {
+        switch(source) {
+            case "Alchemize":
+                if (PotionsObtainedAlchemizeLog.potions_obtained_alchemize != null && AbstractDungeon.floorNum > 0) {
+                    PotionsObtainedAlchemizeLog.potions_obtained_alchemize.get(PotionsObtainedAlchemizeLog.potions_obtained_alchemize.size() - 1).add(potion.ID);
+                }
+                break;
+            case "EntropicBrew":
+                if (PotionsObtainedEntropicBrewLog.potions_obtained_entropic_brew != null && AbstractDungeon.floorNum > 0) {
+                    PotionsObtainedEntropicBrewLog.potions_obtained_entropic_brew.get(PotionsObtainedEntropicBrewLog.potions_obtained_entropic_brew.size() - 1).add(potion.ID);
+                }
+                break;
```

Comment:
With the enum approach and the null check I mentioned, we're able to add a default case:
```
default:
                    logger.warn("Unrecognized potion source: " + source);
                    break;
```

Unlikely to ever be hit but just in case something weird happens (and it's always a good practice to have a default case for a switch statement, to be explicit about what you want the behavior to be).